### ISSUE TITLE
Variables between levels

### ISSUE THREAD

**Issue by Belidos3D** [TheGameCreators/GameGuruRepo#671]
There are many applications for the carrying over of variables between levels, for example an inventory system would need to remember the items it has collected in the previous level, and the easiest way to create an inventory system is by using variables for the items. 

Unfortunately variables, while being remembered in the save/load system, aren't remembered between levels, which makes an inventory system or any system that needs to remember progress from a previous level impossible.

You could get round it by writing scripts to save all your variables at the end of a level and load them in the next level, however that comes with issues, mainly being that if you progressed to a level after collecting items, then loaded a save game on the previous level you would end up with items that you haven't collected because it will load the saved data when you load into the previous saved game.

An ideal fix would if possible be to add to the level switch code the ability for gameguru to carry over any variable prefixed with g_ ? Is this possible?

**Comment by JCLEONITA** [comment]
yes i second that... totally quote.. many of us and many game proejcts needs the ability to carry variables on levels for inventory systema andso no...

hope we can get a fix really soon..

**Comment by GraPhiX-Guru** [comment]
I think this would be possible by creating a text file at the end of each level with a list of each item in the inventory at that time, then when a level is loaded it reads the text file and adds the items to the inventory if you progress to the next level say level 2 a new text file is created with items, so if the user loads a save from level 1 the text file for level 1 is loaded if a save for level to is opened the text for level 2 is applied to the inventory. The text file save could be triggered every time something is added or taken away from the inventory. This method would not cause any memory issues as the memory flush between levels would not affect the text file contents. Just a thought :)

**Comment by JCLEONITA** [comment]
> I think this would be possible by creating a text file at the end of each level with a list of each item in the inventory at that time, then when a level is loaded it reads the text file and adds the items to the inventory if you progress to the next level say level 2 a new text file is created with items, so if the user loads a save from level 1 the text file for level 1 is loaded if a save for level 2 is opened the text for level 2 is applied to the inventory. The text file save could be triggered every time something is added or taken away from the inventory. This method would not cause any memory issues as the memory flush between levels would not affect the text file contents. Just a thought :)
thanks Preben for your reply...
any pratical demostration of that could be possible with some examples for use to use as template to start from?

**Comment by GraPhiX-Guru** [comment]
Ha ha Hi JC I am not Preben, its GraPhiX lol i was just throwing an idea around :)

**Comment by JCLEONITA** [comment]
> Ha ha Hi JC I am not Preben, its GraPhiX lol i was just throwing an idea around :)

ahaha Sorry GraPhix...I cunfused the names.. LOl I know  who you are ..sorry again...

btw i used i similar system like you described in the past to passo over levels variables like hunger, thirst ect.. do you think this could work even with and inventory system?

**Comment by GraPhiX-Guru** [comment]
:) Hi JC
I don't see why not, when you add items to GG in the editor an objectid is created which gets passed out to a map save file so that when you reload your map the items you have placed are loaded exactly where you put them, with a bit of tweaking (and possibly a flag added to the properties panel called "Inventory Item" anything with the flag set to true could be stored in a separate text file ready for GG to load back in, it could either be persistent or temporary (perishes after time like food)  I am not really a coder but I can do this in VB.NET I know because I have done a similar thing in Asset Manager :) for saving fx files and ini files without destroying the original files.

**Comment by JCLEONITA** [comment]
Hi again GraPhix.. and thanks for your explaination...i think if could be doable but i think that gg needs some simple ways to acheive things like that..

**Comment by Belidos3D** [comment]
That wouldn't work, and is the reason smallg's inventory doesn't work across levels, if you had an inventory system like smallg's where your inventory items are entity ID's of entities on the map, you would need the exact same entities with the exact same entity ID on the next map, and how would you tell the next map if that item has already been taken or not?
